src/sentry/static/sentry/app/views/stream/stream.jsx

```diff
@@ -233,22 +231,15 @@ const Stream = createReactClass({
   },
 
   getQueryState(props) {
-    props = props || this.props;
     let currentQuery = props.location.query || {};
-    let state = this.state;
 
     let hasQuery = 'query' in currentQuery;
 
-    let searchId = hasQuery ? null : props.params.searchId || state.searchId || null;
+    let searchId = hasQuery ? null : props.params.searchId || this.state.searchId || null;
 
     let sort = 'sort' in currentQuery ? currentQuery.sort : DEFAULT_SORT;
 
-    let statsPeriod =
-      'statsPeriod' in currentQuery ? currentQuery.statsPeriod : DEFAULT_STATS_PERIOD;
-
-    if (statsPeriod !== '14d' && statsPeriod !== '24h') {
-      statsPeriod = DEFAULT_STATS_PERIOD;
-    }
+    let statsPeriod = STATS_PERIODS.has(currentQuery.statsPeriod) || DEFAULT_STATS_PERIOD;
```

Comment:
actually wait.. do you want this? `statsPeriod` could wind up being `true` right?

Reply:
yikes, thanks, will update both